lib/internal/test_runner/test.js

```diff
@@ -523,12 +523,13 @@ class Test extends AsyncResource {
 
     // The test has run, so recursively cancel any outstanding subtests and
     // mark this test as failed if any subtests failed.
+    this.pendingSubtests = [];
```

Comment:
Should this be using defineProperty in case someone puts a setter on Object.prototype?

Reply:
resolving as this depend on another PR, see https://github.com/nodejs/node/pull/44059#discussion_r933911634